camp_fin/views.py

```diff
@@ -1191,9 +1197,143 @@ def get_context_data(self, **kwargs):
+
+    related_fields = list(filter(lambda x: x.many_to_many is False, related_fields))
+
+    # Loop through all alias objects and migrate their references to the
+    # primary object
+    deleted_objects = []
+    deleted_objects_count = 0
+    for alias_object in alias_objects:
+        # Migrate all foreign key references from alias object to primary
+        # object.
+        for many_to_many_field in many_to_many_fields:
+            alias_varname = many_to_many_field.name
+            related_objects = getattr(alias_object, f"{alias_varname}_set")
+            for obj in related_objects.all():
+                try:
+                    # Handle regular M2M relationships.
+                    getattr(alias_object, alias_varname).remove(obj)
+                    getattr(primary_object, alias_varname).add(obj)
+                except AttributeError:
+                    # Handle M2M relationships with a 'through' model.
+                    # This does not delete the 'through model.
+                    # TODO: Allow the user to delete a duplicate 'through' model.
+                    through_model = getattr(alias_object, alias_varname).through
+                    kwargs = {
+                        many_to_many_field.m2m_reverse_field_name(): obj,
+                        many_to_many_field.m2m_field_name(): alias_object,
+                    }
+                    through_model_instances = through_model.objects.filter(**kwargs)
+                    for instance in through_model_instances:
+                        # Re-attach the through model to the primary_object
+                        setattr(
+                            instance,
+                            many_to_many_field.m2m_field_name(),
+                            primary_object,
+                        )
+                        instance.save()
+                        # TODO: Here, try to delete duplicate instances that are
+                        # disallowed by a unique_together constraint
+
+        for related_field in related_fields:
+            if related_field.one_to_many:
+                alias_varname = related_field.get_accessor_name()
+                related_objects = getattr(alias_object, alias_varname)
+                for obj in related_objects.all():
+                    field_name = related_field.field.name
+                    setattr(obj, field_name, primary_object)
+                    obj.save()
+            elif related_field.one_to_one or related_field.many_to_one:
+                alias_varname = related_field.name
+                related_object = getattr(alias_object, alias_varname)
+                primary_related_object = getattr(primary_object, alias_varname)
+                if primary_related_object is None:
+                    setattr(primary_object, alias_varname, related_object)
+                    primary_object.save()
+                elif related_field.one_to_one:
+                    sys.stdout.write(
```

Comment:
this seems left over from when it was a management command? let's just have this be a print, or nothing since we won't really see it?